tests/PatternAssertions/SocialMediaLinksAssert.php

```diff
@@ -0,0 +1,108 @@
+    self::assertCount(1, $field_list_container);
+  }
+
+  /**
+   * Returns the base CSS selector for a field item depending on the variant.
+   *
+   * @param string $variant
+   *   The variant being checked.
+   *
+   * @return string
+   *   The base selector for the variant.
+   */
+  protected function getBaseItemClass(string $variant): string {
+    if ($variant === 'horizontal') {
+      return '.ecl-social-media-follow.ecl-social-media-follow--horizontal';
+    }
+    return '.ecl-social-media-follow.ecl-social-media-follow--vertical';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getPatternVariant(string $html): string {
+    $crawler = new Crawler($html);
+    $field_list_container = $crawler->filter('div.ecl-social-media-follow');
+    $existing_classes = $field_list_container->attr('class');
+    $existing_classes = explode(' ', $existing_classes);
+    if (in_array('ecl-social-media-follow--horizontal', $existing_classes)) {
+      return 'horizontal';
+    }
+    if (in_array('ecl-social-media-follow--vertical', $existing_classes)) {
+      return 'vertical';
+    }
+    throw new Exception('Variant of social media links pattern is not identified.');
+  }
+
+  /**
+   * Asserts the links of the pattern.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertLinks(array $expected_items, Crawler $crawler): void {
+    $li_items = $crawler->filter('li.ecl-social-media-follow__item');
+    self::assertCount(count($expected_items), $li_items);
+
+    foreach ($expected_items as $index => $expected_item) {
+      $li_item = $li_items->eq($index);
+      if (empty($expected_item['service'])) {
+        // Service icon is absent.
+        $link_element = $label_element = $li_item->filter('a.ecl-link.ecl-link--standalone.ecl-social-media-follow__link');
+      }
+      else {
+        // Service icon exists.
+        $link_element = $li_item->filter('a.ecl-link.ecl-link--standalone.ecl-link--icon.ecl-link--icon-before.ecl-social-media-follow__link');
+        $label_element = $link_element->filter('span.ecl-link__label');
+        $icon_social_file = 'icons-social.svg#';
+        $svg = $link_element->filter('svg.ecl-icon.ecl-icon--xl.ecl-link__icon.ecl-social-media-follow__icon use');
+        self::assertContains($icon_social_file . $expected_item['service'], $svg->attr('xlink:href'));
+        $svg_hover = $link_element->filter('svg.ecl-icon.ecl-icon--xl.ecl-link__icon.ecl-social-media-follow__icon-hover use');
+        self::assertContains($icon_social_file . $expected_item['service'] . '_hover', $svg_hover->attr('xlink:href'));
+      }
+
+      self::assertEquals($expected_item['label'], trim($label_element->text()));
+      self::assertEquals($expected_item['url'], $link_element->attr('href'));
```

Comment:
above suggestions would improve readability and reduce complexity, I imagine it something like:

`
foreach ($expected_items as $index => $expected_item) {
      $li_item = $li_items->eq($index);
      $label_selector = $link_selector = 'a.ecl-link.ecl-link--standalone.ecl-social-media-follow__link';

      if (!empty($expected_item['service'])) {
        $label_selector = 'span.ecl-link__label';
        $svg = $li_item->filter('svg.ecl-icon.ecl-icon--xl.ecl-link__icon.ecl-social-media-follow__icon use');
        self::assertContains('icons-social.svg#' . $expected_item['service'], $svg->attr('xlink:href'));
        $svg_hover = $li_item->filter('svg.ecl-icon.ecl-icon--xl.ecl-link__icon.ecl-social-media-follow__icon-hover use');
        self::assertContains('icons-social.svg#' . $expected_item['service'] . '_hover', $svg_hover->attr('xlink:href'));
      }

      $link_element = $li_item->filter($link_selector);
      $label_element = $li_item->filter($label_selector);

      self::assertEquals($expected_item['label'], trim($label_element->text()));
      self::assertEquals($expected_item['url'], $link_element->attr('href'));
    }
`

Reply:
Such approach doesn't test nesting `svg` inside `a` that is worth, I suppose.

This implementation can be rewritten and support nesting but it requires additional useless invocation of `filter` method that, so I would propose leave current implementation as is.